app/helpers/backend/base_helper.rb

```diff
@@ -292,6 +292,11 @@ def financial_year_selector(financial_year_id = nil, options = {})
       render 'backend/shared/financial_year_selector', financial_year: current_user.current_financial_year, param_name: options[:param_name] || :current_financial_year
     end
 
+    def financial_year_started_on_stopped_on
```

Comment:
Instead of using an helper that only provide a hash to include in HAML tag which is used by JS. 
Use pure `data-` attributes on the targeted input: `data-alert-if-out-of-period` (with alert message as value), `data-period-started-on` (with start date as value) and `data-period-stopped-on` (with finish date as value).
You can have an helper (`current_financial_year_inclusion_attributes`) which will return the attributes directly for `input_html` option of simple form input.
Your JS will detect attributes and build wanted HTML on alert or before (like you prefer).